### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by UnseenFaith**
### Proposed Semver Increment Bump: [MAJOR/MINOR/PATCH]
Minor
### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Changing the structure of Komada to reflect Klasa's structure. This will make building documentation and linting easier.
- Adds Documentation to Komada

Almost all of the documentation is done besides:
- A few more classes (I think it's only settings stuff)
- Only one of the templates (commands) is made. I have yet to make the rest.

### CODE REVIEW

**Review by bdistin — DISMISSED** [review]

File: src/classes/argResolver.js [hunk 1]
```diff
@@ -165,11 +317,21 @@ module.exports = class ArgResolver extends Resolver {
...
     const hyperlink = await super.url(arg);
     if (hyperlink !== null) return hyperlink;
     if (currentUsage.type === "optional" && !repeat) return null;
     throw `${currentUsage.possibles[possible].name} must be a valid url.`;
   }
 
-};
+}
+
+module.exporots = ArgResolver;
```
Comment (kyranet) [inline src/classes/argResolver.js]: lol

File: src/classes/client.js [hunk 2]
```diff
@@ -69,21 +181,56 @@ module.exports = class Komada extends Discord.Client {
...
     const permissions = Discord.Permissions.resolve([...new Set(this.commands.reduce((a, b) => a.concat(b.conf.botPerms), ["READ_MESSAGES", "SEND_MESSAGES"]))]);
     return `https://discordapp.com/oauth2/authorize?client_id=${this.application.id}&permissions=${permissions}&scope=bot`;
   }
 
+  /**
+   * The schema manager for this bot
+   * @readonly
+   * @type {SchemaManager}
+   */
+  get schemaManager() {
```
Comment (kyranet) [inline src/classes/client.js]: This got removed in #313

File: src/classes/loader.js [hunk 3]
```diff
@@ -31,10 +43,23 @@ module.exports = class Loader {
...
+
+    /**
+     * An object containing string paths to piece folders for the user side of Komada
+     * @type {Object}
+     */
     this.clientDirs = makeDirsObject(this.client.clientBaseDir);
   }
 
+  /**
+   * Loads all of the pieces into Komada
```
Comment (kyranet) [inline src/classes/loader.js]: Probs a returns void?

File: src/classes/settingGateway.js [hunk 4]
```diff
@@ -1,205 +1,175 @@
-const SchemaManager = require("./schemaManager");
```
Comment (kyranet) [inline src/classes/settingGateway.js]: This reversed #317 and #322

**Review by kyranet — DISMISSED** [review]
Well, besides that this PR reversed #313 and #322, it looks good to me.

**Review by kyranet — APPROVED** [review]
LGTM

**Pull request merged**